tests/test_extra_files.py

```diff
@@ -48,6 +48,146 @@ def test_slashes(test_case):
 
     assert result == expected
 
+EXTRA_FILE_DOT = {
+    ':': {':': b'a'},
+    ':/': {':': {'index.html': b'a'}},
+    '.nojekyll': {'.nojekyll': b'a'},
+    '/.nojekyll': {'.nojekyll': b'a'},
+    '/info.txt': {'info.txt': b'a'},
+    '/./a/./a.b.': {'a': {'a.b.': b'a'}},
+    '/./a/./a.b./.': {'a': {'a.b.': {'index.html': b'a'}}},
+    './a/./a.b./.': {'a': {'a.b.': {'index.html': b'a'}}},
+    './a/./a.b././': {'a': {'a.b.': {'index.html': b'a'}}},
+    '/./././a.b./.': {'a.b.': {'index.html': b'a'}},
+    'a..b/a../b..': {'a..b': {'a..': {'b..': b'a'}}},
+    'a..b/a../..b/': {'a..b': {'a..': {'..b': {'index.html': b'a'}}}},
+    '/..b../': {'..b..': {'index.html': b'a'}},
+    '/..b..': {'..b..': b'a'},
+    # http: without leading forward slash become absolute url instead of url part
+    'http:': {'http:': b'a'},
+    '/http:': {'http:': b'a'},
+    '/http:/': {'http:': {'index.html': b'a'}},
+    '/https:/': {'https:': {'index.html': b'a'}},
+    '/https://': {'https:': {'index.html': b'a'}},
+    '/http:/https/': {'http:': {'https': {'index.html': b'a'}}},
+}
+@pytest.mark.parametrize('test_case', EXTRA_FILE_DOT)
+def test_dots(test_case):
+    extra_file = {test_case: 'a'}
+    expected = EXTRA_FILE_DOT[test_case]
+    config = {
+        'extra_files': extra_file,
+        'output': {'type': 'dict'},
+    }
+
+    with context_for_test('app_simple') as module:
+        result = freeze(module.app, config)
+
+    # pop to simplify syntax of expected dict
+    # index.html is root page for app_simple, not useful for this test
+    result.pop('index.html')
+
+    assert result == expected
+
+
+EXTRA_FILE_QUOTED = {
+    '%C3%A1/':  {'á': {'index.html': b'a'}},
+    'á/':       {'á': {'index.html': b'a'}},
+    '/a%5cb/':  {'a': {'b': {'index.html': b'a'}}},
```

Comment:
Consider making `%5cb` the way to put a backslash in a filename (which is valid on *nix).
See https://docs.pytest.org/en/6.2.x/skipping.html#skip-xfail-with-parametrize for skipping a parametrized test case on Windows